### ISSUE TITLE
inconsistent number of inlets in [graid] and [grand]

### ISSUE THREAD

**Issue by gilbertohasnofb** [myQwil/pd-externals#1]
Depending on the number of creation arguments, both [graid] and [grand] change their number of inlets.

In both cases, the objects display three inlets when using 2 or less creation arguments, and four inlets when using 3 creation arguments. This creates a problem in the helpfiles of both objects: both of them have a number box labelled "max" which is not connected to neither [graid] nor [grand] and thus does influence them at all.

**Comment by myQwil** [comment]
The help files are still old copies so they need to be fixed, but it is my main intention for the number of inlets to change now based on the number of creation arguments.

**Comment by gilbertohasnofb** [comment]
I see, in that case it would be wonderful to update those help files because the use of those inlets is not obvious (at least not to me), except in the case of 3 creation arguments (in which the 1st inlet receives a bang, and all other three control the 3 arguments).

**Comment by myQwil** [comment]
Okay, the new help files are up for rand, grand, and graid. The dynamic inlet idea was something I debated with myself for a while. If you feel the scale inlet should always be there, just let me know. That was basically just tailored that way for my own personal preference.

**Comment by gilbertohasnofb** [comment]
Miwi, the new help files are now very clear, thanks for the very fast update! 

I personally find the idea of using dynamic inlets a bit confusing, but I do understand where it is coming from. I think it is quite understandable not to have a scale inlet for [grand] since it simply defines the precision of the random numbers (and 100 is a reasonable default), but in the case of [graid] I think this inlet is quite fundamental since it is controlling which values are mapped into the min-max range. It looks a bit strange to me that [graid 5 2] will not create such inlet to control the input value's range and so I have to remember (and use) the default value of 100. 

Here are a couple of other thoughts on these issues:
- [graid] is basically a mapper which takes a value between a and b and maps it into the range c and d. In your case, a is always 0 and b is the default 100, which can be changed only when created with three arguments. Wouldn't it be more interesting to create the object with four possible arguments and respective inlets, so as to completely control the mapping? E.g., mapping input values from 3-13 into the range 100-200.
- I personally feel that the order of the creation arguments and inlets should be min and then max, as opposed to yours max and min. That's also a matter of taste, but I somehow find it a bit unorthodox.
- there is still a small mistake in the help file of [grand]. The word "logic" in the second comment from the top ("random number generator that uses graid logic") collides with the object [graid].

**Comment by myQwil** [comment]
I agree, [graid] should have the scale inlet at all times, whereas [grand] doesn't need it so much.

As far as the order of min and max goes, that was a recent change, which I think should be reverted back for [graid]. For [grand] and [rand], however, I want max to remain as the first creation argument because that way, if there's only one creation argument, it's for max, which means that min will remain 0, as opposed to min being the first argument and max remaining 1.

Also, for the random objects, the order of min and max makes no difference except that [grand] 's second outlet for the raw integer is now in reverse order, but then again, I haven't made much use of the raw integer output thus far, and if it's really necessary, the raw output can be flipped back around with another expression:
[expr (($f1 - min) / -1) + max]
where min is 0 and max is 1000

**Comment by gilbertohasnofb** [comment]
Now I see why the random objects have the order max then min, it makes perfect sense! And I do agree with you that reverting [graid]'s back to min and max could be more intuitive. Everything else sounds great to me, thanks for considering these suggestions of mine.

**Comment by myQwil** [comment]
Okay, [graid] works now so that one creation argument is for max, but two creation arguments are for min, then max. The logic for this was pretty simple so I applied it to [grand] and [rand] and did the builds.

**Comment by gilbertohasnofb** [comment]
Hi miwi, it's working great now, thanks for doing the new builds, I really like the new order of min then max. I also found that the helpfile of [graid] was not updated to this behaviour, so I made the modification myself (simply rewrote a couple of labels) and proposed it to be merged to your repository, as well as a modification of [grand]'s helpfile that had that collision problem I previously mentioned. Thanks a lot for all the work.
